java/src/jmri/jmrit/symbolicprog/tabbedframe/PaneProgFrame.java

```diff
@@ -307,8 +306,7 @@ public boolean doPrompt() {
 
     void setProgrammingGui(JPanel bottom) {
         // see if programming mode is available
-        modePane = getModePane();
-        if (modePane != null) {
+        if (getModePane() != null) {
```

Comment:
No need to change. But it might be good to have:
```
JPanel modePane = getModePane();
if (modePane != null) {
    ...
    temp.add(modePane);
```

The reason is that NPE analysis don't know that `getModePane()` will not change between calls.